### PR TITLE
Alternate simpler fix for jailbreak model loading issue

### PR DESCRIPTION
**Pull request by srampal**
Strip _orig_mod when saving instead of during retrieval as done in prior fix attempt
Alternate fix for https://github.com/redhat-et/semantic_router/pull/54

### CODE REVIEW

**Comment by srampal** [comment]
This fix also includes a self-test that ensures the correct model parameters are saved.

This was verified by fine tuning a model and verifying that the model parameters saved are the same as the original ones without the torch compile prefixes as seen in the output of training below.

{'train_runtime': 131505.762, 'train_samples_per_second': 81.284, 'train_steps_per_second': 3.695, 'train_loss': 0.0005007739498931483, 'epoch': 10.0}
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████| 485870/485870 [36:31:45<00:00,  3.69it/s]
INFO:__main__:Training completed successfully!
INFO:__main__:Detected torch.compile wrapper, extracting original model for saving...
INFO:__main__:Parameter verification: True
INFO:__main__:Model and tokenizer saved to jailbreak_classifier_modernbert-base_model without compile prefixes
INFO:__main__:Evaluating on validation set...
W0824 14:11:52.817000 82749 torch/fx/experimental/symbolic_shapes.py:6823] [2/3] _maybe_guard_rel() was called on non-relation expression Eq(s53, 1) | Eq(s70, s53)
INFO:__main__:Validation accuracy: 0.9998
INFO:__main__:Evaluating on test set...
INFO:__main__:Test accuracy: 0.9998

==================================================
Jailbreak Classification Fine-tuning Completed!
==================================================
Validation Accuracy: 0.9998
Test Accuracy: 0.9998

**Comment by srampal** [comment]
Someone with admin privileges needs to address the workflow permissions error (Owner Notification ...), I believe.

**Comment by Xunzhuo** [comment]
> Author

/cc @rootfs
